### PR TITLE
Fix publishing pre-release versions to AUR

### LINKED ISSUE #3767: Pre-release versions are incorrectly published in AUR
### Describe the bug

User report from: https://aur.archlinux.org/packages/flux-bin#comment-909264

Log from yay:
```
==> Retrieving sources...
  -> Downloading flux-bin-2.0.0rc.1.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     9    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
==> ERROR: Failure while downloading https://github.com/fluxcd/flux2/releases/download/v2.0.0rc.1/flux_2.0.0rc.1_linux_amd64.tar.gz
    Aborting...
```

It looks like the pre-release part of the version is incorrectly parsed.

### Steps to reproduce

Attempt to install flux-bin from AUR on Archlinux

### Expected behavior

The package should reference the correct version.

### Screenshots and recordings

_No response_

### OS / Distro

Archlinux

### Flux version

v2.0.0-rc.1

### Flux check

N/A

### Git provider

_No response_

### Container Registry provider

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by relu**
Fixes #3767

### CODE REVIEW

File: .github/aur/flux-bin/publish.sh [hunk 1]
```diff
@@ -28,20 +28,21 @@ git clone aur@aur.archlinux.org:$PKGNAME $GITDIR 2>&1
 CURRENT_PKGVER=$(cat $GITDIR/.SRCINFO | grep pkgver | awk '{ print $3 }')
 CURRENT_PKGREL=$(cat $GITDIR/.SRCINFO | grep pkgrel | awk '{ print $3 }')
 
-export PKGVER=${VERSION/-/}
+# Transform pre-release to AUR compatible version format
+export PKGVER=${VERSION/-rc./rc}
```
Comment (stefanprodan) [inline .github/aur/flux-bin/publish.sh:32]: So AUR wants `v2.0.0rc1`?

Comment (relu) [reply .github/aur/flux-bin/publish.sh:32]: That's what I've seen in some examples out there (e.g. `1.2.3alpha1` or `. Other options that would work: are `2.0.0rc.1`, `2.0.0.rc1`, or `2.0.0.rc.1`. I couldn't find any official guidelines, the only thing we know for certain is that `-` is reserved as a separator between the version and the package release number.

Comment (stefanprodan) [reply .github/aur/flux-bin/publish.sh:32]: Hmm that’s strange, none of those are valid semver, according to spec

> A pre-release version MAY be denoted by appending a hyphen

Comment (stefanprodan) [reply .github/aur/flux-bin/publish.sh:32]: I would go with `2.0.0rc.1` which preserves the current replacement `${VERSION/-/}`

Comment (relu) [reply .github/aur/flux-bin/publish.sh:32]: True, but I don't think Arch intended to fully support or enforce semver in any way. Here's the section about package versioning from the guidelines: https://wiki.archlinux.org/title/Arch_package_guidelines#Package_versioning

**Review by stefanprodan — APPROVED** [review]
LGTM
Thanks @relu

**Review by hiddeco — APPROVED** [review]

**Pull request merged**